pkg/tests/tasks/traffic/egress/egress_wildcard_hosts_test.go

```diff
@@ -15,76 +15,170 @@
-	test.NewTest(t).Id("T16").Groups(test.Full, test.InterOp).NotRefactoredYet()
-
-	defer cleanupEgressWildcard()
-	defer util.RecoverPanic(t)
-
-	log.Log.Info("Test Egress Wildcard Hosts")
-	sleep := examples.Sleep{Namespace: "bookinfo"}
-	sleep.Install()
-	sleepPod, err := util.GetPodName("bookinfo", "app=sleep")
-	util.Inspect(err, "Failed to get sleep pod name", "", t)
-
-	t.Run("TrafficManagement_egress_direct_traffic_wildcard_host", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Configure direct traffic to a wildcard host")
-		util.KubeApplyContents("bookinfo", EgressWildcardEntry)
-		time.Sleep(time.Duration(10) * time.Second)
-
-		command := `curl -s https://en.wikipedia.org/wiki/Main_Page | grep -o "<title>.*</title>"; curl -s https://de.wikipedia.org/wiki/Wikipedia:Hauptseite | grep -o "<title>.*</title>"`
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-		if strings.Contains(msg, "<title>Wikipedia, the free encyclopedia</title>\n<title>Wikipedia – Die freie Enzyklopädie</title>") {
-			log.Log.Infof("Success. Got Wikipedia response: %s", msg)
-		} else {
-			log.Log.Infof("Error response: %s", msg)
-			t.Errorf("Error response: %s", msg)
-		}
-
-		util.KubeDeleteContents("bookinfo", EgressWildcardEntry)
-	})
-
-	t.Run("TrafficManagement_egress_gateway_wildcard_host", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Configure egress gateway to a wildcard host")
-		util.KubeApplyContents("bookinfo", util.RunTemplate(EgressWildcardGatewayTemplate, smcp))
-		time.Sleep(time.Duration(10) * time.Second)
-
-		command := `curl -s https://en.wikipedia.org/wiki/Main_Page | grep -o "<title>.*</title>"; curl -s https://de.wikipedia.org/wiki/Wikipedia:Hauptseite | grep -o "<title>.*</title>"`
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-		if strings.Contains(msg, "<title>Wikipedia, the free encyclopedia</title>\n<title>Wikipedia – Die freie Enzyklopädie</title>") {
-			log.Log.Infof("Success. Got Wikipedia response: %s", msg)
-		} else {
-			log.Log.Infof("Error response: %s", msg)
-			t.Errorf("Error response: %s", msg)
-		}
-
-		util.KubeDeleteContents("bookinfo", util.RunTemplate(EgressWildcardGatewayTemplate, smcp))
+	NewTest(t).Id("T16").Groups(Full, InterOp).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		ns := "bookinfo"
+
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns, EgressWildcardGatewayTemplate, EgressWildcardEntry)
+		})
+
+		t.Log("This Test recieves the sleep pod name")
+		app.InstallAndWaitReady(t, app.Sleep(ns))
+		t.LogStep("Recieve the sleep pod name")
+		retry.UntilSuccess(t, func(t TestHelper) {
+			oc.Exec(t,
+				pod.MatchingSelector(ns, "app=sleep"),
+				"sleep",
+				ns,
+				assert.OutputContains(
+					"",
+					"Successfully got the sleep pod name",
+					"Failed to get the sleep pod name"))
```

Comment:
This checks if the output contains "", which is always the case. Not sure what the purpose of this assertion is. Actually, the whole `oc.Exec` command doesn't make sense, since it passes `ns` as the command. This will always fail, since there is no command called `"bookinfo"`.